converters/provider-aws/dynamodb/dynamodb.go

```diff
@@ -47,44 +47,49 @@ func TableResource(mg resource.Managed) ([]resource.Managed, error) {
 			target.Spec.ForProvider.RangeKey = t.AttributeName
 		}
 	}
-	for _, t := range source.Spec.ForProvider.GlobalSecondaryIndexes {
-		parameter := &targetv1beta1.GlobalSecondaryIndexParameters{
-			Name:             t.IndexName,
-			NonKeyAttributes: t.Projection.NonKeyAttributes,
-			ProjectionType:   t.Projection.ProjectionType,
-		}
-		for _, a := range t.KeySchema {
-			if *a.KeyType == "HASH" {
-				parameter.HashKey = a.AttributeName
-			}
-			if *a.KeyType == "RANGE" {
-				parameter.RangeKey = a.AttributeName
+	if source.Spec.ForProvider.GlobalSecondaryIndexes != nil {
+		for _, t := range source.Spec.ForProvider.GlobalSecondaryIndexes {
+			if t.IndexName != nil {
+				parameter := &targetv1beta1.GlobalSecondaryIndexParameters{
+					Name:             t.IndexName,
+					NonKeyAttributes: t.Projection.NonKeyAttributes,
+					ProjectionType:   t.Projection.ProjectionType,
+				}
+				for _, a := range t.KeySchema {
+					if *a.KeyType == "HASH" {
+						parameter.HashKey = a.AttributeName
+					}
+					if *a.KeyType == "RANGE" {
+						parameter.RangeKey = a.AttributeName
+					}
+				}
+				target.Spec.ForProvider.GlobalSecondaryIndex = append(target.Spec.ForProvider.GlobalSecondaryIndex, *parameter)
 			}
 		}
-		target.Spec.ForProvider.GlobalSecondaryIndex = append(target.Spec.ForProvider.GlobalSecondaryIndex, *parameter)
 	}
-
-	for _, t := range source.Spec.ForProvider.LocalSecondaryIndexes {
-		parameter := &targetv1beta1.LocalSecondaryIndexParameters{
-			Name:             t.IndexName,
-			NonKeyAttributes: t.Projection.NonKeyAttributes,
-			ProjectionType:   t.Projection.ProjectionType,
-		}
-		for _, a := range t.KeySchema {
-			if *a.KeyType == "RANGE" {
-				parameter.RangeKey = a.AttributeName
+	if source.Spec.ForProvider.LocalSecondaryIndexes != nil {
```

Comment:
Do we really need this check for this slice?

Reply:
@sergenyalcin this slice can be null?